### PR TITLE
Show warning for debug builds on devices

### PR DESCRIPTION
**Pull request by yschimke**
image

With a temporary edit to flip isEmulator

### CODE REVIEW

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt [hunk 1]
```diff
@@ -61,6 +75,24 @@ fun CustomTimeText(
...
+                        text = debugWarning!!,
+                        style = CurvedTextStyle(textStyle),
+                        color = Color.Red
+                    )
+                }
+            } else null,
+            trailingLinearContent = if (showWarning) {
+                {
+                    Text(
+                        text = debugWarning!!,
```
Comment (yschimke) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:90]: This is to workaround a compiler bug

image

**Comment by LouisCAD** [comment]
I'd add "(slow)" or change the message to indicate that performances are expected to be poor because it lacks release optimizations.

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt [hunk 2]
```diff
@@ -39,6 +43,16 @@ fun CustomTimeText(
     leadingText: String
 ) {
     val textStyle = TimeTextDefaults.timeTextStyle()
+    val debugWarning = remember {
+        val isEmulator = Build.PRODUCT.startsWith("sdk_gwear")
+
+        if (BuildConfig.DEBUG && !isEmulator) {
+            "Debug (Slow) Build"
```
Comment (BowerSteve) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:50]: Better to keep the text short, how about "Debug (slower)"

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt [hunk 3]
```diff
@@ -61,6 +75,24 @@ fun CustomTimeText(
                     )
                 }
             } else null,
+            trailingCurvedContent = if (showWarning) {
```
Comment (BowerSteve) [inline ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:78]: How about adding a comment that trailing text is against UX guidance for real apps, but is convenient here as a reminder not to use Debug builds for performance testing?

Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:78]: Do we have a link to the guidance?

Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:78]: Will update once we find it.

File: ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt (hunk 1 above)
Comment (yschimke) [reply ComposeAdvanced/app/src/main/java/com/example/android/wearable/composeadvanced/presentation/components/CustomTimeText.kt:91]: https://youtrack.jetbrains.com/issue/KT-51838

**Pull request merged**